### PR TITLE
Refactor the testing folder

### PR DESCRIPTION
**Pull request by RaunoArike**
- Moves files in `src/test` into separate folders
- Splits `src/test/counterfactuals.jl` into multiple files
- Fixes a bug in the implementation of the Feature Tweak generator encountered while testing

### CODE REVIEW

File: test/generators/counterfactuals.jl [hunk 1]
```diff
@@ -0,0 +1,142 @@
+include("feature_tweak.jl")
+include("generators.jl")
+include("probe.jl")
```
Comment (kmariuszk) [inline test/generators/counterfactuals.jl:3]: Aren't we supposed to add those include/using statements in the `runtests.jl` file?

**Review by kmariuszk — CHANGES_REQUESTED** [review]

File: test/generators/counterfactuals.jl (hunk 1 above)
Comment (RaunoArike) [reply test/generators/counterfactuals.jl:3]: We can do it both ways, but my current structure is that there's one main test file in each separate folder that is included in the `runtests.jl` file and this main file includes other test files from the same folder. I'm fine with including all of them in the `runtests.jl` file as well if you prefer that, it's just that the list of files there would get rather long.

File: test/generators/generators.jl [hunk 2]
```diff
@@ -1,12 +1,7 @@
-using CounterfactualExplanations
-using CounterfactualExplanations.Data
-using CounterfactualExplanations.Generators
-using Random
-
 @testset "Construction" begin
     @testset "Generic" begin
         generator = GenericGenerator()
```
Comment (laurikskl) [inline test/generators/generators.jl:3]: The generators could be called from Generators as well I believe? You've done this in a couple of places

File: test/models/generative_models.jl [hunk 3]
```diff
@@ -8,10 +6,12 @@ ys = counterfactual_data.y
 generator = generator_catalogue[:revise]()
```
Comment (laurikskl) [inline test/models/generative_models.jl:6]: the catalogue could be referenced from Generators I believe

File: test/other/plotting.jl [hunk 4]
```diff
@@ -1,23 +1,22 @@
-using MLUtils
-using Plots
-
 generator = generator_catalogue[:generic]()
```
Comment (laurikskl) [inline test/other/plotting.jl:1]: Here as well?

**Review by laurikskl — CHANGES_REQUESTED** [review]

File: test/generators/generators.jl (hunk 2 above)
Comment (RaunoArike) [reply test/generators/generators.jl:3]: Right, will fix this

File: test/models/generative_models.jl (hunk 3 above)
Comment (RaunoArike) [reply test/models/generative_models.jl:6]: Oh right, forgot that the catalogue comes from there

**Review by laurikskl — APPROVED** [review]
LGTM, thanks for the changes

**Pull request merged**